### PR TITLE
LLL: further improve error reporting

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: liblll/CodeFragment.h [hunk 1]
```diff
@@ -51,6 +51,11 @@ class CodeFragment
 	void finalise(CompilerState const& _cs);
 
 	template <class T> void error() const { BOOST_THROW_EXCEPTION(T() ); }
+	template <class T> void error(std::string reason) const {
```
Comment (chriseth) [inline liblll/CodeFragment.h]: `const&`?

File: liblll/Parser.cpp [hunk 2]
```diff
@@ -135,10 +135,17 @@ void dev::eth::parseTreeLLL(string const& _s, sp::utree& o_out)
 			s.push_back(i);
 	}
 	auto ret = s.cbegin();
-	qi::phrase_parse(ret, s.cend(), element, space, qi::skip_flag::dont_postskip, o_out);
+	try {
```
Comment (chriseth) [inline liblll/Parser.cpp]: Brace to the next line.

File: liblll/Parser.cpp [hunk 3]
```diff
@@ -135,10 +135,17 @@ void dev::eth::parseTreeLLL(string const& _s, sp::utree& o_out)
...
+		qi::phrase_parse(ret, s.cend(), element, space, qi::skip_flag::dont_postskip, o_out);
+	}
+	catch(const qi::expectation_failure<it>& e)
```
Comment (chriseth) [inline liblll/Parser.cpp]: Space after `catch` and `const` should be right before `&`

**Review by chriseth — CHANGES_REQUESTED** [review]

**Comment by axic** [comment]
@chriseth changes are done.

**Review by chriseth — APPROVED** [review]

**Pull request merged**